admin/modules/config/settings.php

```diff
@@ -927,6 +927,21 @@
 			$forum_group_select[] = $name;
 		}
 
+		// Administrator is changing the login method.
+		if($mybb->settings['username_method'] == 1 || $mybb->settings['username_method'] == 2 || $mybb->input['upsetting']['username_method'] == 1 || $mybb->input['upsetting']['username_method'] == 2)
+		{
+			$query = $db->simple_select('users', 'email, COUNT(email) AS duplicates', "email!=''", array('group_by' => 'email HAVING duplicates>1'));
+			if($db->num_rows($query))
+			{
+				$mybb->input['upsetting']['username_method'] = 0;
+			}
+			else
+			{
+				$mybb->input['upsetting']['allowmultipleemails'] = 0;
+			}
+			$lang->success_settings_updated .= $lang->success_settings_updated_username_method;
```

Comment:
Can't use the same error for both cases.

`success_settings_updated_username_method` says that `Allowed Login Methods` wasn't updated due to duplicate Emails, there should be another error for the other setting.